behave_django/runner.py

```diff
@@ -0,0 +1,62 @@
+try:
+    from django.test.runner import DiscoverRunner
+except ImportError:
+    from django.test.simple import DjangoTestSuiteRunner as DiscoverRunner
+
+from django.core.management import call_command
+
+try:
+    from django.shortcuts import resolve_url
+except ImportError:
+    # support Django 1.4, which has no resolve_url() yet
+    from django.shortcuts import redirect
+
+    resolve_url = lambda to, *args, **kwargs: \
+        redirect(to, *args, **kwargs)['Location']
+
+from behave_django.testcase import (BehaviorDrivenTestCase,
+                                    ExistingDatabaseTestCase)
+
+
+class BehaviorDrivenTestRunner(DiscoverRunner):
+    testcase_class = BehaviorDrivenTestCase
+
+    def before_scenario(self, context):
```

Comment:
The problem started with that there was no elegant way to provide the `optparse` options in places outside the behave management command module. Also, I had two places where I had to make an `if`-`else` with the options, the TestRunner and the TestCase, because both contain logic that affect the database (setup and fixture-teardown).

In the end this was how I could have all the logic that belongs together in one place. The clutter is less of a disadvantage when you think that it's easy to understand how the parts play together. In the end the `environment` module was obsolete because of monkey-patching.

We could write a Mixin, maybe, to tear the hooks out into a separate file, but I'd have to try and see whether this makes things clearer or more fuzzy. Test case and test runner are now even called `BehaviorDriven...`, isn't that cute?